soc/arm/nordic_nrf/nrf52/soc_power.h warning spew when CONFIG_SYS_POWER_MANAGEMENT=n

**Describe the bug**

The above mentioned soc_power.h declares function prototypes which use `enum power_states`, provided by power.h.

However, power.h only defines this enum if CONFIG_SYS_POWER_MANAGEMENT=y. Otherwise, the enum is undefined and leads to the following warnings emitted when building any .c file that includes soc_power.h, directly or indirectly:

```
/home/mbolivar/zephyrproject/zephyr/soc/arm/nordic_nrf/nrf52/soc_power.h:29:31: warning: ‘enum power_states’ declared inside parameter list will not be visible outside of this definition or declaration
 void sys_set_power_state(enum power_states state);
                               ^~~~~~~~~~~~
/home/mbolivar/zephyrproject/zephyr/soc/arm/nordic_nrf/nrf52/soc_power.h:34:36: warning: ‘enum power_states’ declared inside parameter list will not be visible outside of this definition or declaration
 void sys_power_state_post_ops(enum power_states state);
                                    ^~~~~~~~~~~~
```

This is a *lot* of files, so it would be nice to avoid this issue when power management is disabled by providing alternate prototypes that don't cause warnings.

**To Reproduce**
Steps to reproduce the behavior:
1. Build any application which has CONFIG_SYS_POWER_MANAGEMENT=n. (I used the Zephyr SDK 0.9.5 in case the gcc warning output is version-specific)
2. Look at warnings in build output

**Expected behavior**
No warnings

**Impact**
Lots and lots of warnings :)

**Screenshots or console output**
See above

**Environment (please complete the following information):**
 - OS: all
 - Toolchain: any recent-ish gcc should do
 - Commit SHA: f2ef52f12240cd891aa471e5984fd22a3a1f5e93


**Additional context**
N/A

--- a/include/power.h
+++ b/include/power.h
@@ -14,10 +14,6 @@
 extern "C" {
 #endif
 
-#ifdef CONFIG_SYS_POWER_MANAGEMENT
-
-extern unsigned char sys_pm_idle_exit_notify;
-
 /**
  * @defgroup power_management_api Power Management
  * @{
@@ -55,6 +51,10 @@ enum power_states {
 #endif /* CONFIG_SYS_POWER_DEEP_SLEEP */
 	SYS_POWER_STATE_MAX
 };
+
+#ifdef CONFIG_SYS_POWER_MANAGEMENT
+
+extern unsigned char sys_pm_idle_exit_notify;
 
 /**
  * @brief System Power Management API

--- a/include/power.h
+++ b/include/power.h
@@ -8,6 +8,7 @@
 #define ZEPHYR_INCLUDE_POWER_H_
 
 #include <zephyr/types.h>
+#include <stdbool.h>
 
 #ifdef __cplusplus
 extern "C" {

